src/main/kotlin/uk/gov/gdx/datashare/services/AcquirersService.kt

```diff
@@ -131,6 +150,48 @@ class AcquirersService(
     }
   }
 
+  fun deleteAcquirerSubscription(subscriptionId: UUID): AcquirerSubscription {
+    val now = dateTimeHandler.now()
+    adminActionAlertsService.noticeAction(
+      AdminAction(
+        "Delete acquirer subscription",
+        object {
+          val subscriptionId = subscriptionId
+          val whenDeleted = now
+        },
+      ),
+    )
+    val subscription = acquirerSubscriptionRepository.save(
+      acquirerSubscriptionRepository.findByIdOrNull(subscriptionId)?.copy(
+        whenDeleted = now,
+      ) ?: throw AcquirerSubscriptionNotFoundException("Subscription $subscriptionId not found"),
+    )
+
+    acquirerSubscriptionEnrichmentFieldRepository
+      .findAllByAcquirerSubscriptionId(subscriptionId)
+      .forEach { deleteAcquirerSubscriptionEnrichmentField(it.id) }
+
+    if (subscription.oauthClientId != null) {
+      val otherSubscriptionsWithClient =
+        acquirerSubscriptionRepository.findAllByOauthClientId(subscription.oauthClientId)
+      if (otherSubscriptionsWithClient.isEmpty()) {
+        cognitoService.deleteUserPoolClient(subscription.oauthClientId)
+      }
+      return subscription
+    }
+
+    if (subscription.queueName == null) {
+      throw IllegalStateException("Acquirer does not have a client id or queue name.")
+    }
+
```

Comment:
This doesn't feel like it's in quite the right place, but if I do the check on queueName and oauthClientId before the cognito deletion, kotlin doesn't pick pick up that the queueName is not null in the if statement below